src/python/gudhi/cubical_complex.py

```diff
@@ -276,8 +158,153 @@ def persistence_intervals_in_dimension(self, dimension):
         :returns: The persistence intervals.
         :rtype:  numpy array of dimension 2
 
-        :note: intervals_in_dim function requires :func:`compute_persistence` function to be
-            launched first.
+        :note: intervals_in_dim function requires :func:`~gudhi.CubicalComplex.compute_persistence`
+            or :func:`persistence` function to be launched first.
+        """
+        assert self._pcohptr != None, "compute_persistence() or persistence() must be called before persistence_intervals_in_dimension()"
+        return np.array(self._pcohptr.intervals_in_dimension(dimension))
+
+class CubicalComplex(CubicalComplexBaseClass):
+    """The CubicalComplex is an example of a structured complex useful in
+    computational mathematics (specially rigorous numerics) and image
+    analysis.
+    """
+
+    def __init__(self, dimensions=None, top_dimensional_cells=None,
+                 perseus_file=''):
+        """CubicalComplex constructor from dimensions and
+        top_dimensional_cells or from a Perseus-style file name.
+
+        :param dimensions: A list of number of top dimensional cells.
+        :type dimensions: list of int
+        :param top_dimensional_cells: A list of cells filtration values.
+        :type top_dimensional_cells: list of double
+
+        Or
+
+        :param top_dimensional_cells: A multidimensional array of cells
+            filtration values.
+        :type top_dimensional_cells: anything convertible to a numpy ndarray
+
+        Or
+
+        :param perseus_file: A Perseus-style file name.
+        :type perseus_file: string
+        """
+        if ((dimensions is not None) and (top_dimensional_cells is not None)
+            and (perseus_file == '')):
+            CubicalComplexBaseClass._thisptr = Cubical_complex_interface_(dimensions, top_dimensional_cells)
```

Comment:
The name `_thisptr` looks a bit strange now, but I guess it doesn't hurt.

Reply:
I renamed attributes members as it is no more a pointer on 1b816100